Mathlib/Algebra/BigOperators/Finprod.lean

```diff
@@ -1007,6 +1021,15 @@ theorem finsum_mul {R : Type*} [Semiring R] (f : α → R) (r : R) (h : (support
     (∑ᶠ a : α, f a) * r = ∑ᶠ a : α, f a * r :=
   (AddMonoidHom.mulRight r).map_finsum h
 
+@[to_additive (attr := simp)]
+lemma finprod_apply {α ι : Type*} {f : ι → α → N} (hf : (mulSupport f).Finite) (a : α) :
```

Comment:
I guess if only `(mulSupport (f i)).Finite` for each `i`, then you run into trouble when `ι`, and the LHS is a junk value while the RHS is not...